src/components/content/heroes/aurora/weather-api.ts

```diff
@@ -1,17 +1,65 @@
 import axios from 'axios';
 
 const API_KEY = process.env.WEATHER_API_KEY;
-const latitude = 48.1351;
-const longitude = 11.582;
+const BASE_URL = 'https://api.weatherapi.com/v1';
 
 export async function getWeather() {
-  const apiUrl = `https://api.weatherapi.com/v1/current.json?key=${API_KEY}&q=${latitude},${longitude}`;
+  try {
+    const response = await axios.get(
+      `${BASE_URL}/ip.json?key=${API_KEY}&q=auto:ip`,
+    );
+    const { lat, lon } = response.data;
+    return getWeatherOfLocation(lat, lon);
+  } catch (error) {
+    console.error('Error fetching IP geolocation:', error);
+    return 'NotSet';
+  }
+}
+
+export async function getWeatherOfLocation(latitude, longitude) {
+  const apiUrl = `${BASE_URL}/current.json?key=${API_KEY}&q=${latitude},${longitude}`;
 
   try {
     const response = await axios.get(apiUrl);
-    return response.data.current.condition.text;
+    const conditionCode = response.data.current.condition.code;
+    return getWeatherDescription(conditionCode);
   } catch (error) {
     console.error('Error fetching weather:', error);
-    throw new Error('Unable to fetch weather information.');
+    return 'NotSet';
+  }
+}
+
+function getWeatherDescription(conditionCode) {
+  switch (conditionCode) {
+    case 1000:
+      return 'Sunny';
+    case 1003:
+    case 1006:
+      return 'Cloudy';
+    case 1063:
+    case 1180:
+    case 1183:
+    case 1186:
+    case 1189:
+    case 1192:
+    case 1195:
+    case 1198:
+    case 1201:
+    case 1204:
+    case 1207:
+    case 1240:
+    case 1243:
+      return 'Rainy';
+    case 1114:
+    case 1117:
+    case 1210:
+    case 1213:
+    case 1216:
+    case 1219:
+    case 1222:
+    case 1225:
+      return 'Snowy';
+    default:
+      return 'NotSet';
```

Comment:
Nice cool decision.

But you are leaving out some parameters. I guess those are the "condition edge cases" such as "patchy light drizzle" :)

See this List of all possible weather condition codes (with german names): [reference](https://www.weatherapi.com/docs/conditions.json)
![image](https://github.com/virus-rpi/satellytes.com/assets/78978542/e973490e-9769-4588-b295-5dd0807c98ff)


I agree that it is difficult to define our weathertype for some of these conditions (eg is 1150 cloudy or already rainy?), but it is better to show a somehow not totally weather than none, right? So please include the missing condition codes too.

And also, as you can see in the screenshot, the condition codes are sorted ascending by code and from sunny to snowy. What do you think about defining some ranges (eg. `if code => 1009 && code =< 1150` for cloudy) to map the code to our WeatherType. This can avoid a huge switch.

And, as above, please use WeatherType.Cloudy etc. directly here for correct Typing.